steampy/utils.py

```diff
@@ -72,6 +84,63 @@ def merge_items(items: List[dict], descriptions: dict, **kwargs) -> dict:
     return merged_items
 
 
+def get_market_listings_from_html(html: str) -> dict:
+    document = BeautifulSoup(html, "html.parser")
+    nodes = document.select("div[id=myListings]")[0].findAll("div", {"class": "market_home_listing_table"})
+    sell_listings_dict = {}
+    buy_orders_dict = {}
+    for node in nodes:
+        if "My sell listings" in node.text:
+            sell_listings_dict = get_sell_listings_from_node(node)
+        elif "My listings awaiting confirmation" in node.text:
+            sell_listings_awaiting_conf = get_sell_listings_from_node(node)
+            for listing in sell_listings_awaiting_conf.values():
+                listing["need_confirmation"] = True
+            sell_listings_dict.update(sell_listings_awaiting_conf)
+        elif "My buy orders" in node.text:
+            buy_orders_dict = get_buy_orders_from_node(node)
+    return {"buy_orders": buy_orders_dict, "sell_listings": sell_listings_dict}
+
+
+def get_sell_listings_from_node(node: Tag) -> dict:
+    sell_listings_raw = node.findAll("div", {"id": re.compile('mylisting_\d+')})
+    sell_listings_dict = {}
+    for listing_raw in sell_listings_raw:
+        spans = listing_raw.select("span[title]")
+        listing = {
+            "listing_id": listing_raw.attrs["id"].replace("mylisting_", ""),
+            "buyer_pay": spans[0].text.strip(),
+            "you_receive": spans[1].text.strip()[1:-1],
+            "created_on": listing_raw.findAll("div", {"class": "market_listing_listed_date"})[0].text.strip(),
+            "need_confirmation": False
+        }
+        sell_listings_dict[listing["listing_id"]] = listing
+    return sell_listings_dict
+
+
+def get_buy_orders_from_node(node: Tag) -> dict:
+    buy_orders_raw = node.findAll("div", {"id": re.compile('mybuyorder_\\d+')})
+    buy_orders_dict = {}
+    for order in buy_orders_raw:
+        qnt_price_raw = order.select("span[class=market_listing_price]")[0].text.split("@")
+        order = {
+            "order_id": order.attrs["id"].replace("mybuyorder_", ""),
+            "quantity": int(qnt_price_raw[0].strip()),
+            "price": qnt_price_raw[1].strip(),
+            "item_name": order.a.text
+        }
+        buy_orders_dict[order["order_id"]] = order
+    return buy_orders_dict
+
+
+def get_listing_id_to_assets_address_from_html(html: str) -> dict:
+    listing_id_to_assets_address = {}
+    regex = "CreateItemHoverFromContainer\( [\w]+, 'mylisting_([\d]+)_[\w]+', ([\d]+), '([\d]+)', '([\d]+)', [\d]+ \);"
+    for match in re.findall(regex, html):
+        listing_id_to_assets_address[match[0]] = [str(match[1]), match[2], match[3]]
+    return listing_id_to_assets_address
+
+
 def get_description_key(item: dict) -> str:
```

Comment:
Utils is package for some methods to be use many times, i think you can move these new methods to market module/class. or if market module/class would be to long maybe make market directory and inside it helpers module with those functions